File: PracticasProgramacion/Practica_09/cpp/EstadisticasNumeros.cpp
#include "EstadisticasNumeros.h"
#include <iostream>
#include <algorithm>
#include <numeric>
EstadisticasNumeros::EstadisticasNumeros()
    : numeros_{0}, suma_(0), promedio_(0), mediana_(0), maximo_(0), minimo_(0) {}

void EstadisticasNumeros::ingresarNumeros() {
    std::cout << "Ingrese 5 numeros:\n";
    for (int i = 0; i < 5; ++i) {
        std::cout << "Numero " << (i+1) << ": ";
        std::cin >> numeros_[i];
    }
    calcular();
}

void EstadisticasNumeros::calcular() {
    suma_ = std::accumulate(std::begin(numeros_), std::end(numeros_), 0.0);
    promedio_ = suma_ / 5.0;
    double copia[5];
    for (int i = 0; i < 5; ++i) copia[i] = numeros_[i];
    std::sort(std::begin(copia), std::end(copia));
    mediana_ = copia[2];
    maximo_ = *std::max_element(std::begin(numeros_), std::end(numeros_));
    minimo_ = *std::min_element(std::begin(numeros_), std::end(numeros_));
}

void EstadisticasNumeros::mostrarResultados() const {
    std::cout << "\nSuma: " << suma_ << "\n";
    std::cout << "Promedio: " << promedio_ << "\n";
    std::cout << "Mediana: " << mediana_ << "\n";
    std::cout << "Maximo: " << maximo_ << "\n";
    std::cout << "Minimo: " << minimo_ << "\n";
}

double EstadisticasNumeros::getSuma() const { return suma_; }
double EstadisticasNumeros::getPromedio() const { return promedio_; }
double EstadisticasNumeros::getMediana() const { return mediana_; }
double EstadisticasNumeros::getMaximo() const { return maximo_; }
double EstadisticasNumeros::getMinimo() const { return minimo_; }

File: PracticasProgramacion/Practica_09/python/estadisticas_numeros.py
"""
EstadisticasNumeros: 5 valores, cálculo de suma, promedio, mediana, max, min.
Mediana = valor central tras ordenar; para n=5 es el índice 2.
"""


class EstadisticasNumeros:
    N = 5

    def __init__(self) -> None:
        self._numeros = [0.0] * self.N
        self._suma = 0.0
        self._promedio = 0.0
        self._mediana = 0.0
        self._maximo = 0.0
        self._minimo = 0.0

    def ingresar_numeros(self) -> None:
        print("Ingrese 5 numeros:")
        for i in range(self.N):
            self._numeros[i] = float(input(f"Numero {i+1}: "))
        self._calcular()

    def _calcular(self) -> None:
        self._suma = sum(self._numeros)
        self._promedio = self._suma / self.N
        copia = sorted(self._numeros)
        self._mediana = copia[self.N // 2]
        self._maximo = max(self._numeros)
        self._minimo = min(self._numeros)

    def mostrar_resultados(self) -> None:
        print(f"\nSuma: {self._suma}")
        print(f"Promedio: {self._promedio}")
        print(f"Mediana: {self._mediana}")
        print(f"Maximo: {self._maximo}")
        print(f"Minimo: {self._minimo}")
